package/kernel/linux/modules/netfilter.mk

```diff
@@ -1190,3 +1253,25 @@ define KernelPackage/nft-compat
 endef
 
 $(eval $(call KernelPackage,nft-compat))
+
+define KernelPackage/nft-socket
+  SUBMENU:=$(NF_MENU)
+  TITLE:=Netfilter nf_tables socket support
+  DEPENDS:=+kmod-nft-core +kmod-nf-socket +kmod-nf-socket6
+  FILES:=$(foreach mod,$(NFT_SOCKET-m),$(LINUX_DIR)/net/$(mod).ko)
+  AUTOLOAD:=$(call AutoProbe,$(notdir $(NFT_SOCKET-m)))
+  KCONFIG:=$(KCONFIG_NFT_SOCKET)
+endef
+
+$(eval $(call KernelPackage,nft-socket))
+
+define KernelPackage/nft-tproxy
+  SUBMENU:=$(NF_MENU)
+  TITLE:=Netfilter nf_tables tproxy support
+  DEPENDS:=+kmod-nft-core +kmod-nf-defrag +kmod-nf-defrag6 +kmod-nf-tproxy +kmod-nf-tproxy6
+  FILES:=$(foreach mod,$(NFT_TPROXY-m),$(LINUX_DIR)/net/$(mod).ko)
+  AUTOLOAD:=$(call AutoProbe,$(notdir $(NFT_TPROXY-m)))
+  KCONFIG:=$(KCONFIG_NFT_TPROXY)
+endef
+
+$(eval $(call KernelPackage,nft-tproxy))
```

Comment:
I think you should separate tproxy and socket into ipv4 and ipv6 versions.

Reply:
```
config NFT_TPROXY
	tristate "Netfilter nf_tables tproxy support"
	depends on IPV6 || IPV6=n
	select NF_DEFRAG_IPV4
	select NF_DEFRAG_IPV6 if NF_TABLES_IPV6
	select NF_TPROXY_IPV4
	select NF_TPROXY_IPV6 if NF_TABLES_IPV6
	help
	  This makes transparent proxy support available in nftables.
```
Only nft-tproxy exists in the kernel, not nft-tproxy-ipv4 and nft-tproxy-ipv6, and even then do we have to separate nft-tproxy and nft-tproxy6?
Of course, nft-socket is too.

Reply:
Yeah, but if you see there is a line `select NF_TPROXY_IPV6 if NF_TABLES_IPV6` that selects NF_TPROXY_IPV6 if IPv6 is enabled, either you could do something like this +@IPV6:NF_TPROXY_IPV6 or you can separate them, choose whichever you think is correct.
Someday I'll have to standardize the netfilter config makefile, but only for nftables/when iptables is removed as this is a complete mess

Reply:
Once I added `+@IPV6:` to the dependencies, the other package dependencies were fine.

Thanks again for the help given.